test/jit/test_cuda.py

```diff
@@ -97,14 +97,14 @@ def test_current_stream(self):
         @torch.jit.script
         def fn():
             device_index = torch.cuda.current_device()
-            s0 = torch.cuda.current_stream(device_index)
-            s1 = torch.cuda.current_stream(1)
-            s2 = torch.cuda.current_stream(0)
+            device = torch.device("cuda:" + str(device_index))
+            s0 = torch.cuda.current_stream(device)
+            s1 = torch.cuda.current_stream(torch.device("cuda:1"))
```

Comment:
This doesn't look very backwards compatible

Reply:
Nobody is using this feature yet (they are waiting for this diff to land first) so I don't think BC is a problem. However, the eager mode `current_stream` can accept `device`, `int`, or `str`, so we need to do that in order to get API parity. I initially thought it would not be possible due to the lack of a `Union` type in JIT, so I thought supporting `device` arguments would make the most sense.

However, this API is desugared and not compiled, so I think multiple JIT operators can be registered for it with different input types. I commented elsewhere in the PR on how to do this.

Reply:
Oh, so just to make sure I understand you correctly, both int and device work in eager mode, but in the current version of this diff, only device works inside torch.jit.script? I agree this is less bad then.

Reply:
Yes.